File: src/main/java/aurocosh/divinefavor/common/lib/SimpleCounter.java
package aurocosh.divinefavor.common.lib;

public class SimpleCounter {
    private int required;
    private int count;

    public SimpleCounter(int required) {
        this.required = required;
        reset();
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }

    public void setRequired(int required) {
        this.required = required;
        reset();
    }

    public void reset() {
        count = 0;
    }

    public boolean count() {
        if(count >= required)
            return true;
        count++;
        return false;
    }

    public boolean isFinished() {
        return count >= required;
    }
}

File: src/main/java/aurocosh/divinefavor/common/lib/SimpleCounter.kt
package aurocosh.divinefavor.common.lib

open class SimpleCounter(private var required: Int) {
    var count: Int = 0

    val isFinished: Boolean
        get() = count >= required

    init {
        reset()
    }

    open fun setRequired(required: Int) {
        this.required = required
        reset()
    }

    fun reset() {
        count = 0
    }

    fun count(): Boolean {
        if (count >= required)
            return true
        count++
        return false
    }
}
